### ISSUE TITLE
Rule Change: func-style declaration to allow arrow functions with let variables

### ISSUE THREAD

**Issue by jacekkopecky** [eslint/eslint#18570]
### What rule do you want to change?

func-style

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

```js
let a = () => 2;
a = () => 3;
console.log(a);
```

### What does the rule currently do for this code?

With the setting `/*eslint func-style: ["error", "declaration"]*/`, the rule complains about the `let a=...` line. The declaration style is (erroneously, in my opinion) applied to variables declared with `let`. 

The value of a `let` variable can be changed (and should be, or we should use `const`), so the line cannot be replaced by a function declaration.

### What will the rule do after it's changed?

The rule should allow `let` variables to be assigned function expressions despite the preferred style being "declaration".

I'm not sure if this should be the new default behaviour, a new option, or an additional value `"let"` for the existing option `allowArrowFunctions`, with which we could have:

```js
/*eslint func-style: ["error", "declaration", { "allowArrowFunctions": "let" }]*/

let a = () => 2;   // this would now be valid
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by Rec0iL99** [comment]
Hi @jacekkopecky, thanks for the issue. As per our rules policies [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we are not adding new options to the `func-style` rule. You can always clone the rule into a custom rule and make the preferred changes as you wish.

**Comment by Rec0iL99** [comment]
Plus, the `allowArrowFunctions` expects a [boolean](https://github.com/eslint/eslint/blob/ff6e96ec30862a4eb77a201551ec8c618335bfc2/lib/rules/func-style.js#L30) value. I think this rule setting solves the issue you are facing.

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgZnVuYy1zdHlsZTogW1wiZXJyb3JcIiwgXCJkZWNsYXJhdGlvblwiLCB7IFwiYWxsb3dBcnJvd0Z1bmN0aW9uc1wiOiB0cnVlIH1dKi9cblxubGV0IGEgPSAoKSA9PiAyO1xuYSA9ICgpID0+IDM7XG5jb25zb2xlLmxvZyhhKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)

**Comment by jacekkopecky** [comment]
Thanks @Rec0iL99, I missed that policy in reading the how-to on proposing new options ealier today...
I wish there were a list of community plugins where people might contribute this, so a wider audience can benefit from these types of improvements.

`allowArrowFunctions` allows const expressions as well which is something our code style wants to discourage.

**Comment by jacekkopecky** [comment]
@Rec0iL99 in particular, I looked at https://eslint.org/docs/latest/contribute/propose-rule-change and around, which suggests that making rules more configurable will be considered. It would be useful to make the policy about stylistic rules more visible, I guess.

**Comment by Rec0iL99** [comment]
Feel free to open a PR to improve the docs 👍
